sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala

```diff
@@ -62,7 +62,9 @@ case class SortExec(
     "peakMemory" -> SQLMetrics.createSizeMetric(sparkContext, "peak memory"),
     "spillSize" -> SQLMetrics.createSizeMetric(sparkContext, "spill size"))
 
-  private[sql] var rowSorter: UnsafeExternalRowSorter = _
+  // Each task has its own instance of UnsafeExternalRowSorter. It is created in the
+  // createSorter method and stored in a ThreadLocal variable.
+  private[sql] var rowSorter: ThreadLocal[UnsafeExternalRowSorter] = _
```

Comment:
To be more clear, this won't affect current Spark usage. We hit this when we're developing something that accesses this in parallel. I think it is better to have it more robust code in general.